examples/bitbucket/role.yaml

```diff
@@ -33,4 +33,38 @@ subjects:
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: Role
-  name: tekton-triggers-bitbucket-minimal
\ No newline at end of file
+  name: tekton-triggers-bitbucket-minimal
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
```

Comment:
Instead of creating a new ClusterRoleBinding and Role, can we modify the existing Role to add the "list", and "watch" verbs? 
https://github.com/tektoncd/triggers/blob/master/examples/bitbucket/role.yaml#L16

In general, we'd want the ServiceAccount for the EventListener to have as few permissions as it can. With a ClusterRole, it will have access to all resources(triggers, bindings, etc.) in the cluster while with a Role it will only have access to the triggers within the resources within the namespace where it was created (this should be sufficient for our use cases).

Reply:
Yes, I tried. But got this error `Failed to list *v1alpha1.TriggerTemplate: triggertemplates.triggers.tekton.dev is forbidden: User "system:serviceaccount:default:tekton-triggers-github-sa" cannot list resource "triggertemplates" in API group "triggers.tekton.dev" at the cluster scope` for every kind of resource that is involved, e.g. Trigger, TriggerTemplate, and TriggerBinding. 
Looks like the lister is used at the cluster scope. Do you mean the Lister should not be used at the cluster scope? I will take a look.

Reply:
sorry missed this comment earlier. I think the issue here is that we are using a cluster scoped informer. We are setting up the informers here:  https://github.com/tektoncd/triggers/pull/821/files#diff-255de153352c6a922d11a7a76e0655d6dee57016f105177e0fbb7774bc1ae8b6R118


Instead what we can do is manually setup the Informers and restrict them to a namespace using  https://github.com/tektoncd/triggers/blob/v0.9.1/pkg/client/informers/externalversions/factory.go#L92

Something like:
```
	factory := informerfactory.NewSharedInformerFactoryWithOptions(clientset, 30*time.Second, informerfactory.WithNamespace(namespace)) // we can pass in the EL namespace here

```

Once we have the factory, we'll also have to start it:

```
	go func(ctx context.Context) {
		factory.Start(ctx.Done())
		<-ctx.Done()
	}(ctx)
```

and then return the Informers as we do now: ` factory.Triggers().V1alpha1().EventListeners(),`

Reply:
ClusterTriggerBinding is a cluster scoped informer anyway. Should we exclude ClusterTriggerBinding when creating the Sink object? or it will emit the error without creating a new ClusterRoleBinding and ClusterRole.

Reply:
ahh for ClusterTriggerBinding, we'll need a cluster scoped informer. For the rest, we can get away with a namespace scoped one.

Reply:
Thanks for explanation! This is what I'm doing currently. I was not sure it was proper or not before.